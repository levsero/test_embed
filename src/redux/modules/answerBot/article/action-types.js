export const ARTICLE_DISMISSED_PENDING = 'widget/answerBot/ARTICLE_DISMISSED_PENDING';
export const ARTICLE_DISMISSED_FULFILLED = 'widget/answerBot/ARTICLE_DISMISSED_FULFILLED';
export const ARTICLE_DISMISSED_REJECTED = 'widget/answerBot/ARTICLE_DISMISSED_REJECTED';

export const ARTICLE_VIEWED_PENDING = 'widget/answerBot/ARTICLE_VIEWED_PENDING';
export const ARTICLE_VIEWED_FULFILLED = 'widget/answerBot/ARTICLE_VIEWED_FULFILLED';
export const ARTICLE_VIEWED_REJECTED = 'widget/answerBot/ARTICLE_VIEWED_REJECTED';
