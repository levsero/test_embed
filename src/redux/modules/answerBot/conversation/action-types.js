export const QUESTION_VALUE_CHANGED = 'widget/answerBot/QUESTION_VALUE_CHANGED';

export const QUESTION_SUBMITTED_PENDING = 'widget/answerBot/QUESTION_SUBMITTED_PENDING';
export const QUESTION_SUBMITTED_FULFILLED = 'widget/answerBot/QUESTION_SUBMITTED_FULFILLED';
export const QUESTION_SUBMITTED_REJECTED = 'widget/answerBot/QUESTION_SUBMITTED_REJECTED';

export const CONVERSATION_SCROLL_CHANGED = 'widget/answerBot/CONVERSATION_SCROLL_CHANGED';
export const CONVERSATION_SCREEN_CLOSED = 'widget/answerBot/CONVERSATION_SCREEN_CLOSED';
