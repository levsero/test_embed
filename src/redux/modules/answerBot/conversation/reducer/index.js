import { combineReducers } from 'redux';

import lastScroll from './last-scroll';
import lastScreenClosed from './last-screen-closed';

export default combineReducers({
  lastScroll,
  lastScreenClosed
});
