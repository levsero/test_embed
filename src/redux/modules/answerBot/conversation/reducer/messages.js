import {
  QUESTION_SUBMITTED_PENDING,
  QUESTION_SUBMITTED_FULFILLED
} from '../action-types';

import {
  BOT_MESSAGE,
  BOT_CHANNEL_CHOICE,
  BOT_FEEDBACK,
  BOT_FEEDBACK_REQUESTED
} from '../../root/action-types';

const initialState = new Map();

let messageCount = 0;

const addMessage = (messages, message) => {
  const copy = new Map(messages);

  return copy.set(messageCount++, { ...message });
};

const messages = (state = initialState, action) => {
  const { type, payload } = action;

  switch (type) {
    case QUESTION_SUBMITTED_PENDING:
      return addMessage(state, { ...payload, isVisitor: true });
    case QUESTION_SUBMITTED_FULFILLED:
      const {
        message, // eslint-disable-line no-unused-vars
        ...rest
      } = payload;

      return addMessage(state, { ...rest, type: 'results' });
    case BOT_MESSAGE:
      return addMessage(state, payload);
    case BOT_CHANNEL_CHOICE:
      return addMessage(state, { ...payload, type: 'channelChoice' });
    case BOT_FEEDBACK:
      return addMessage(state, { ...payload, isVisitor: true, type: 'feedback' });
    case BOT_FEEDBACK_REQUESTED:
      return addMessage(state, { ...payload, type: 'feedbackRequested' });
    default:
      return state;
  }
};

export default messages;
