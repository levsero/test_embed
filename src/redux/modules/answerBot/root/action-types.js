export const BOT_MESSAGE = 'widget/answerBot/BOT_MESSAGE';
export const BOT_GREETED = 'widget/answerBot/BOT_GREETED';
export const BOT_INITIAL_FALLBACK = 'widget/answerBot/BOT_INITIAL_FALLBACK';
export const BOT_CHANNEL_CHOICE = 'widget/answerBot/BOT_CHANNEL_CHOICE';
export const BOT_FEEDBACK = 'widget/answerBot/BOT_FEEDBACK';
export const BOT_FEEDBACK_REQUESTED = 'widget/answerBot/BOT_FEEDBACK_REQUESTED';

export const SCREEN_CHANGED = 'widget/answerBot/SCREEN_CHANGED';
export const ARTICLE_SHOWN = 'widget/answerBot/ARTICLE_SHOWN';

export const INPUT_DISABLED = 'widget/answerBot/INPUT_DISABLED';
