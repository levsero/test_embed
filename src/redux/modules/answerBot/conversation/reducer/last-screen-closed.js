import { CONVERSATION_SCREEN_CLOSED } from '../action-types'

const initialState = 0

const lastScreenClosed = (state = initialState, action) => {
  switch (action.type) {
    case CONVERSATION_SCREEN_CLOSED:
      return action.payload
    default:
      return state
  }
}

export default lastScreenClosed
