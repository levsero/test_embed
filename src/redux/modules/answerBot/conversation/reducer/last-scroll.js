import { CONVERSATION_SCROLL_CHANGED } from '../action-types';

const initialState = 0;

const lastScroll = (state = initialState, action) => {
  switch (action.type) {
    case CONVERSATION_SCROLL_CHANGED:
      return action.payload;
    default:
      return state;
  }
};

export default lastScroll;
