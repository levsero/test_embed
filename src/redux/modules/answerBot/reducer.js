import { combineReducers } from 'redux';

import messages from './conversation/reducer/messages';
import conversation from './conversation/reducer';
import sessions from './sessions/reducer/sessions';
import currentMessage from './root/reducer/current-message';
import currentSessionID from './root/reducer/current-session-id';
import currentArticle from './root/reducer/current-article';
import currentContextualArticle from './root/reducer/current-contextual-article';
import currentScreen from './root/reducer/current-screen';
import greeted from './root/reducer/greeted';
import contextualSearchFinished from './root/reducer/contextual-search-finished';
import initialFallbackSuggested from './root/reducer/initial-fallback-suggested';

export default combineReducers({
  messages,
  conversation,
  currentMessage,
  sessions,
  currentSessionID,
  currentArticle,
  currentScreen,
  greeted,
  initialFallbackSuggested,
  contextualSearchFinished,
  currentContextualArticle
});
