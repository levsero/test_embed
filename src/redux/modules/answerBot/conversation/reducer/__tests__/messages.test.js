import reducer from '../messages'
import * as actionTypes from '../../action-types'
import * as rootActionTypes from 'src/redux/modules/answerBot/root/action-types'

const matchesSnapshot = state => {
  expect(Array.from(state.values())).toMatchSnapshot()
}

test('initial state is empty map', () => {
  expect(reducer(undefined, { type: '' }).size).toEqual(0)
})

test('QUESTION_VALUE_SUBMITTED updates to expected state', () => {
  const payload = {
    timestamp: 123456,
    message: 'Surely not EVERYONE was kung fu fighting.'
  }

  const state = reducer(undefined, {
    type: actionTypes.QUESTION_VALUE_SUBMITTED,
    payload
  })

  matchesSnapshot(state)
})

test('QUESTION_SUBMITTED_FULFILLED updates to expected state', () => {
  const payload = {
    timestamp: 123456,
    message: [
      { id: 1, title: 'article1' },
      { id: 2, title: 'article2' },
      { id: 3, title: 'article3' }
    ]
  }

  const state = reducer(undefined, {
    type: actionTypes.QUESTION_SUBMITTED_FULFILLED,
    payload
  })

  matchesSnapshot(state)
})

test('BOT_FEEDBACK_REQUESTED updates to expected state', () => {
  const state = reducer(undefined, {
    type: rootActionTypes.BOT_FEEDBACK_REQUESTED
  })

  matchesSnapshot(state)
})

test('BOT_FEEDBACK updates to expected state', () => {
  const payload = {
    timestamp: 123456
  }

  const state = reducer(undefined, {
    type: rootActionTypes.BOT_FEEDBACK,
    payload
  })

  matchesSnapshot(state)
})

test('BOT_CHANNEL_CHOICE updates to expected state', () => {
  const state = reducer(undefined, {
    type: rootActionTypes.BOT_CHANNEL_CHOICE
  })

  matchesSnapshot(state)
})

test('BOT_TYPING updates to expected state', () => {
  const state = reducer(undefined, {
    type: rootActionTypes.BOT_TYPING,
    payload: {
      timestamp: 123456
    }
  })

  matchesSnapshot(state)
})
