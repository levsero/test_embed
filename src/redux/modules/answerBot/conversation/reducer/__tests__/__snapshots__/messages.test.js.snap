// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BOT_CHANNEL_CHOICE updates to expected state 1`] = `
Array [
  Object {
    "type": "channelChoice",
  },
]
`;

exports[`BOT_FEEDBACK updates to expected state 1`] = `
Array [
  Object {
    "isVisitor": true,
    "timestamp": 123456,
    "type": "feedback",
  },
]
`;

exports[`BOT_FEEDBACK_REQUESTED updates to expected state 1`] = `
Array [
  Object {
    "type": "feedbackRequested",
  },
]
`;

exports[`BOT_TYPING updates to expected state 1`] = `
Array [
  Object {
    "timestamp": 123456,
    "type": "botTyping",
  },
]
`;

exports[`QUESTION_SUBMITTED_FULFILLED updates to expected state 1`] = `
Array [
  Object {
    "timestamp": 123456,
    "type": "results",
  },
]
`;

exports[`QUESTION_VALUE_SUBMITTED updates to expected state 1`] = `
Array [
  Object {
    "isVisitor": true,
    "message": "Surely not EVERYONE was kung fu fighting.",
    "timestamp": 123456,
  },
]
`;
